File: projects/lostengine/lost/rg/ClearColor.cpp
#include "lost/rg/ClearColor.h"
#include "lost/gl/Context.h"

namespace lost
{
namespace rg
{

ClearColor::ClearColor(const common::Color& inCol)
{
  col = inCol;
  name = "clearColor";
}

ClearColorPtr ClearColor::create(const common::Color& inCol)
{
  return ClearColorPtr(new ClearColor(inCol));
}

void ClearColor::process(gl::ContextPtr ctx)
{
  if(!active) return;
  ctx->clearColor(col);
}

}
}

File: projects/lostengine/lost/lua/lost/io/rg/ClearColor.lua
module("lost.io", package.seeall)

require("lost.io.Loader")

-- calls the glClear command with the given mask
function Loader:ClearColor(definition)
  local color = definition["color"]
  if color == nil then
    error("color is required for rg:ClearColor")
  end
  local result = lost.rg.ClearColor.create(color)
  local active = definition["active"]
  local name = definition["name"]
  if active ~= nil then
    result.active = active
  end
  if name ~= nil then
    result.name = name
  end
  return result
end